airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -2341,6 +2356,104 @@ definitions:
       $parameters:
         type: object
         additionalProperties: true
+  AsyncJobStatusMap:
+    description: Matches the api job status to Async Job Status.
+    type: object
+    required:
+      - running
+      - completed
+      - failed
+      - timeout
+    properties:
+      type:
+        type: string
+        enum: [AsyncJobStatusMap]
+      running:
+        type: array
+        items:
+          type: string
+      completed:
+        type: array
+        items:
+          type: string
+      failed:
+        type: array
+        items:
+          type: string
+      timeout:
+        type: array
+        items:
+          type: string
+  AsyncRetriever:
+    description: Retrieves records by Asynchronously sending requests to fetch records. The retriever acts as an orchestrator between the requester, the record selector, the paginator, and the partition router.
+    type: object
+    required:
+      - type
+      - record_selector
+      - status_mapping
+      - creation_requester
+      - polling_requester
+      - download_requester
+    properties:
+      type:
+        type: string
+        enum: [AsyncRetriever]
+      record_selector:
+        description: Component that describes how to extract records from a HTTP response.
+        "$ref": "#/definitions/RecordSelector"
+      status_mapping:
+        description: Async Job Status to Airbyte CDK Async Job Status mapping.
+        anyOf:
+          - "$ref": "#/definitions/AsyncJobStatusMap"
+      status_extractor:
+        description: Responsible for fetching the actual status of the async job.
+        anyOf:
+          - "$ref": "#/definitions/CustomRecordExtractor"
```

Comment:
It is a bit annoying that if we add an extractor, we need to update all the places where we reference extractors. Same thing with the requesters, decoders, etc... Do we have a solution for that?

Reply:
unfortunately not to my knowledge :/

Reply:
can you create a reference to the anyOf?